setup_utils.py

```diff
@@ -0,0 +1,149 @@
+import ast
+import os
+import re
+import subprocess
+
+
+_R_RC = re.compile("rc(\d+)$")
+
+
+def parse_version(path):
+    with open(path) as fp:
+        return _AssignmentParser().parse(fp.read())["version"]
+
+
+def write_version_py(filename, major, minor, micro, is_released,
+                     previous_version=None):
```

Comment:
Is the `previous_version` argument here just in case we need it later? It's not currently used.